components/script/dom/htmlinputelement.rs

```diff
@@ -1110,7 +1122,8 @@ impl VirtualMethods for HTMLInputElement {
         if let Some(ref s) = self.super_type() {
             s.bind_to_tree(tree_in_doc);
         }
-
+        // sanitize when it is first parsed
+        self.sanitize_value();
```

Comment:
This doesn't look right. Here's [the part of the spec](https://html.spec.whatwg.org/multipage/input.html#attr-input-value) which defines this behaviour (emphasis mine):

> The value content attribute gives the default value of the input element. When the value content attribute is added, set, or removed, **if the control's dirty value flag is false**, the user agent must set the value of the element to the value of the value content attribute, if there is one, **or the empty string otherwise**, and then run the current value sanitization algorithm, if one is defined.